### DISCUSSION TITLE
tests: develop and write tests for steps/incremental_clustering.py

### DISCUSSION THREAD

**Discussion by anujsinha3** [uw-ssec/MAWpy#102] [Ideas]
The Incremental Clustering code has been refactored from using CSV files to the pandas data frame.

The task is to develop and write tests for the code at: src/mawpy/steps/incremental_clustering.py 

To write tests:

- create a folder called 'steps' under the 'tests' folder
- In the above folder, create a file called 'test_incremental_clustering.py'
- define test functions corresponding to each function src/mawpy/steps/incremental_clustering.py

**Comment by qzchen-uw** [comment]
If a user starts with a csv file, they can also use this code, right? in other words, they do not need to first pre-process it into pandas data frame before processing it? Also, is the code limited by the data size?

**Comment by anujsinha3** [comment]
Yes, they can use the csv file. The code will take care of conversion the csv file into pandas dataframe.

More than the code itself, it depends on the machine that is running the code about its data handling capacity.
